server/statistics/hot_cache.go

```diff
@@ -190,29 +173,30 @@ func (w *HotCache) updateItems(queue <-chan FlowItemTask, runTask func(task Flow
 	}
 }
 
-func (w *HotCache) runReadTask(task FlowItemTask) {
+func (w *HotCache) runReadTask(task flowItemTask) {
 	if task != nil {
-		// TODO: do we need a run-task timeout to protect the queue won't be stucked by a task?
-		task.runTask(w.readFlow)
-		hotCacheFlowQueueStatusGauge.WithLabelValues(Read.String()).Set(float64(len(w.readFlowQueue)))
+		// TODO: do we need a run-task timeout to protect the queue won't be stuck by a task?
+		task.runTask(w.readCache)
+		hotCacheFlowQueueStatusGauge.WithLabelValues(Read.String()).Set(float64(len(w.readCache.taskQueue)))
 	}
 }
 
-func (w *HotCache) runWriteTask(task FlowItemTask) {
+func (w *HotCache) runWriteTask(task flowItemTask) {
 	if task != nil {
-		// TODO: do we need a run-task timeout to protect the queue won't be stucked by a task?
-		task.runTask(w.writeFlow)
-		hotCacheFlowQueueStatusGauge.WithLabelValues(Write.String()).Set(float64(len(w.writeFlowQueue)))
+		// TODO: do we need a run-task timeout to protect the queue won't be stuck by a task?
+		task.runTask(w.writeCache)
+		hotCacheFlowQueueStatusGauge.WithLabelValues(Write.String()).Set(float64(len(w.writeCache.taskQueue)))
 	}
 }
 
-func update(item *HotPeerStat, flow *hotPeerCache) {
-	flow.Update(item)
-	if item.IsNeedDelete() {
-		incMetrics("remove_item", item.StoreID, item.Kind)
-	} else if item.IsNew() {
+func updateStat(cache *hotPeerCache, item *HotPeerStat) {
```

Comment:
how about `updateCache`?

Reply:
It will be updated in https://github.com/tikv/pd/pull/4484